analyses/malawi/flooding_trigger/08_mwi_event_visualization.md

```diff
@@ -0,0 +1,130 @@
+### Visualizing flood events with GloFAS streamflow
+
+This notebook visualizes GloFAS streamflow (historical and forecasted) during dates when a historical flood event has been identified by an external dataset. The objective of this is to better understand the quality of the external event dataset and general agreement with GloFAS streamflow. 
+
+```python
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+import numpy as np
+import pandas as pd
+import os
+from pathlib import Path
+import sys
+import matplotlib.dates as mdates
+
+from matplotlib.ticker import MaxNLocator
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[1]}/"
+sys.path.append(path_mod)
+
+from src.indicators.flooding.config import Config
+from src.indicators.flooding.glofas import utils
+
+config = Config()
+mpl.rcParams['figure.dpi'] = 300
+
+PLOT_DIR = config.DATA_DIR / 'processed' / 'mwi' / 'plots' / 'flooding'
+PRIVATE_DIR = config.DATA_PRIVATE_DIR
+EXPLORE_DIR = PRIVATE_DIR / 'exploration' / 'mwi' / 'flooding'
+
+LEADTIMES = [x + 1 for x in range(10)]
+SAVE_PLOT = False
+EVENT = 'RCO' # 'rco', 'floodscan', 'combined'
+COUNTRY_ISO3 = 'mwi'
+
+stations_adm2 = {
+    #'G1724': 'Nsanje',
+    'G2001': 'Chikwawa'
+}
+```
+
+Read in the processed GloFAS data.
+
+```python
+ds_glofas_reanalysis = utils.get_glofas_reanalysis(
+    country_iso3=COUNTRY_ISO3)
+ds_glofas_reforecast = utils.get_glofas_reforecast(
+    country_iso3 = COUNTRY_ISO3, leadtimes=LEADTIMES,
+    interp=True
+)
+ds_glofas_reforecast_summary = utils.get_glofas_forecast_summary(ds_glofas_reforecast)
+df_return_period = utils.get_return_periods(ds_glofas_reanalysis, method='analytical')
+```
+
+Read in the processed event dataset(s). Here, we're using the combined RCO events from both Chikwawa and Nsanje. 
+
+```python
+events = {}
+for station in stations_adm2.values():
+    events[station] = pd.read_csv(EXPLORE_DIR / f'all_rco_event_summary.csv')
+```
+
+Create a plot for each event. For the dates around each flood onset, we want to see what both the GloFAS reanalysis (historical) and reforecast data look like. We'll pick a specific forecast probability and include forecasts from various leadtimes. The y-axis is scaled to the percentage of a given return period threshold. Lines on the graph have a stronger weight when the values exceed the threshold.
+
+```python
+buffer_start = 5
```

Comment:
It might be good to increase this, since it's helpful to know if the forecast predicts something early, as that could be considered a true positive